src/subtype.c

```diff
@@ -3328,6 +3339,13 @@ static jl_value_t *intersect(jl_value_t *x, jl_value_t *y, jl_stenv_t *e, int pa
     return jl_bottom_type;
 }
 
+static jl_value_t *simple_meet2(jl_value_t *a, jl_value_t *b) {
+    // make sure A=B A=C would be merged into A<:Union{B, C}.
```

Comment:
```
477  // in many cases, we need to over-estimate this by returning `b`.
```
IIUC, for this call to `simple_meet`, this comment should say "in many cases, we need to under-estimate this by returning `Union{}`", and then swap all of the cases accordingly?